tests/main.test.js

```diff
@@ -0,0 +1,129 @@
+import path from 'node:path'
+import url from 'node:url'
+import {promises as fs} from 'node:fs'
+import {
+  run,
+  updatecliDownload,
+  updatecliVersion,
+  updatecliExtract,
+} from 'src/main'
+
+const directory = path.dirname(url.fileURLToPath(import.meta.url))
+
+const cachePath = path.join(directory, 'CACHE')
+const temporaryPath = path.join(directory, 'TEMP')
+// Set temp and tool directories before importing (used to set global state)
+process.env['RUNNER_TEMP'] = temporaryPath
+process.env['RUNNER_TOOL_CACHE'] = cachePath
+
+const version = 'v0.28.0'
```

Comment:
Could potentially read the version from the `action.yml`

Reply:
Or added as target in this manifest: https://github.com/updatecli/updatecli-action/blob/v2/updatecli/updatecli.d/updatecli.yaml#L36-L43

Reply:
added target in aae16d3